tests/datadriven/test_reads.py

```diff
@@ -24,7 +59,58 @@ def getDataModelClass(self):
     def getProtocolClass(self):
         return protocol.GAReadGroupSet
 
-    def testFixMe(self):
+    def testGetReadAlignmentsBothRefs(self):
+        # test that querying by both referenceName and referenceId fails
+        with self.assertRaises(exceptions.BadReadsSearchRequestBothRefs):
+            readGroupSet = self._gaObject
+            for readGroup in readGroupSet._readGroups:
+                list(readGroup.getReadAlignments("a", 5, 0, None))
+
+    def testGetReadAlignments(self):
+        # test that searching with no arguments succeeds
+        readGroupSet = self._gaObject
+        for i, readGroup in enumerate(readGroupSet._readGroups):
+            alignments = list(readGroup.getReadAlignments(
+                None, None, None, None))
+            for j, alignment in enumerate(alignments):
+                otherAlignment = self._reads[i][j]
+                self.assertAlignmentsEqual(alignment, otherAlignment)
+
+    def testGetReadAlignmentsRefName(self):
+        # test that searching with a reference name succeeds
         readGroupSet = self._gaObject
-        # TODO
-        self.assertIsNotNone(readGroupSet)
+        for i, readGroup in enumerate(readGroupSet._readGroups):
+            refNames = self._refNames[i]
+            refName = refNames.keys()[0]
+            refCount = refNames[refName]
+            alignments = list(readGroup.getReadAlignments(
+                refName, None, None, None))
+            self.assertEqual(len(alignments), refCount)
+
+    def testGetReadAlignmentsRefId(self):
+        # test that searching with a reference id succeeds
+        readGroupSet = self._gaObject
+        for i, readGroup in enumerate(readGroupSet._readGroups):
+            refIds = self._refIds[i]
+            refId = refIds.keys()[0]
+            refCount = refIds[refId]
+            alignments = list(readGroup.getReadAlignments(
+                None, refId, None, None))
+            self.assertEqual(len(alignments), refCount)
+
+    def testGetReadAlignmentsStartEnd(self):
+        # test that searching with start and end coords succeeds
+        readGroupSet = self._gaObject
+        for i, readGroup in enumerate(readGroupSet._readGroups):
+            refNames = self._refNames[i]
+            refName = refNames.keys()[0]
+            refCount = refNames[refName]
+            bigNumThatPysamWontChokeOn = 2**30
+            alignments = list(readGroup.getReadAlignments(
+                refName, None, 0, bigNumThatPysamWontChokeOn))
+            self.assertEqual(len(alignments), refCount)
+
+    def assertAlignmentsEqual(self, readOne, readTwo):
```

Comment:
It would be good to be a bit clearer (what's readOne and readTwo?). I think gaRead and pysamRead is the most unambiguous way of stating it. Also, we'll want a lot more asserts here --- we should check every attribute on the GA read to make sure it is correct.